pkg/providers/ack/ack.go

```diff
@@ -148,8 +151,19 @@ func (p *DefaultProvider) resolveUserData(respStr string,
 	updatedCommand = fmt.Sprintf("%s --node-config %s", updatedCommand, cfg)
 
 	// Add taints
-	taint := karpv1.UnregisteredNoExecuteTaint
-	updatedCommand = fmt.Sprintf("%s --taints %s", updatedCommand, taint.ToString())
+	taints := lo.Flatten([][]corev1.Taint{
+		nodeClaim.Spec.Taints,
+		nodeClaim.Spec.StartupTaints,
```

Comment:
StartupTaints was added by karpenter, not us(provider).

Reply:
So it need to be removed here?

Reply:
> That's too weird, although karpenter is compatible with this approach. Could it be for compatibility with the non-pod class initializer on nodes?

Could you please explain `what's the non-pod class initializer on nodes`.

Reply:
Example: A binary to perform initialization work on the node. It starts with the system, not the container. When it is done working, it needs to remove the StartupTaint. In this scenario, if Karpenter adds StartupTaint, binary is likely to miss taint. Therefore, taint and kubelet need to be started together.

This is what I can think of where it would apply.

Reply:
I checked the logic is building script(with taints) first, and then create instance using ecs api, so I think the binary can get the taint some show from user data after execution completed, so the binary will not miss it, correct me if I'm wrong.

Reply:
Sorry, I didn't make it clear. But it doesn't matter, this is fine for now.

Please fix error in presubmit.

Reply:
Fixed